tests/benchmark_msbar_evolution.py

```diff
@@ -125,3 +146,102 @@ def benchmark_APFEL_msbar_evolution(self):
                 np.testing.assert_allclose(
                     apfel_vals, np.sqrt(np.array(my_vals)), rtol=5e-3
                 )
+
+    def benchmark_APFEL_msbar_solution(self):
+        apfel_vals_dict = {
+            "EXA": {
+                0: np.array(
+                    [1.9855, 4.8062, 175.0000],
+                ),
+                1: np.array(
+                    [2.1308, 4.9656, 175.0000],
+                ),
+                2: np.array([2.1566, 4.9841, 175.0000]),
+            },
+        }
+        # collect my values
+        for order in [0, 1, 2]:
+            theory_dict["PTO"] = order
+            my_masses = compute_msbar_mass(theory_dict)
+            # get APFEL numbers - if available else use cache
+            apfel_vals = apfel_vals_dict[theory_dict["ModEv"]][order]
+            if use_APFEL:
+                # run apfel
+                apfel.CleanUp()
+                apfel.SetTheory("QCD")
+                apfel.SetPerturbativeOrder(order)
+                apfel.SetAlphaEvolution("exact")
+                apfel.SetAlphaQCDRef(theory_dict["alphas"], theory_dict["Qref"])
+                apfel.SetVFNS()
+                apfel.SetMSbarMasses(
+                    theory_dict["mc"], theory_dict["mb"], theory_dict["mt"]
+                )
+                apfel.SetMassScaleReference(
+                    theory_dict["Qmc"], theory_dict["Qmb"], theory_dict["Qmt"]
+                )
+                apfel.SetRenFacRatio(theory_dict["fact_to_ren_scale_ratio"])
+                apfel.InitializeAPFEL()
+                apfel.EnableWelcomeMessage(1)
```

Comment:
of course you need to recompute `apfek_vals` here - as we're doing here: https://github.com/N3PDF/eko/blob/d02dbe839f02d43915f54650fe3dbad83a583d0b/tests/benchmark_strong_coupling.py#L130-L134

Reply:
So I was not able to extract the values of the computed quark masses with an apfel function... so I just copied them from the log message manually. I know it's an hack, the proper way would be to have https://github.com/N3PDF/external/blob/afd30d08f95ed96e4b03df21e0ecc1d284a7292e/apfel/src/Evolution/MSbarmass.f#L478 available in the python wrapper.

Reply:
okay, I see ... then I guess we can not do it